### PR TITLE
Feature/4152 upgrade to node 14

### LINKED ISSUE #4152: Upgrade Node to latest
### Summary

We use Node to build our site.
We're currently using Node 10.16.0 (and npm 6.10.2).
According to the [release schedule](https://nodejs.org/en/about/releases/):

- Node 10.16.0 is currently in maintenance mode and will reach its end of life (no more updates) in a few months (30 April 2021)
- Node 12 is active but will enter maintenance mode next month (30 November 2020) and end of life in April 2023
- Node 14 is current and will start LTS (long-term support) tomorrow (27 October 2020)
- Node 16 isn't expected until April 2021
- Node isn't a fan of odd-numbered releases

Because
- our current version is nearly out of date,
- the next version is current,
- but the next-next version will be current probably by the time you read this, and
- because our current CircleCI image defaults to a newer version of Node and we specifically roll it back,

I propose we upgrade from 10 to 14 (skipping 12).

Upgrading should keep us up to date with capabilities and reported vulnerabilities

### PR DESCRIPTION
**Pull request by rfultz**
## Summary

- Resolves #4152 
- Will upgrade Node from 10 to 14
- Updates required package.json entry
- Updates optional package.json entries
- Updates package-lock.json
- Updates CircleCI configuration to use Node 14 instead of 10

## Impacted areas of the application

Node runs everything involved in building the front-end of the site, touching the CSS, JavaScript, images, fonts, automated tests, file naming…

## Screenshots

Should be no visible changes

## Related PRs

None

## Reviewers
Need a thorough front-end review and probably a back-end double-check (thinking CircleCI config, mostly). Feel free to add or remove reviewers.

## How to test

- Pull the branch like normal
- Start virtual environment
- Check your version of Node `nvm ls`. There should be a `->     v14.13.0` listed. If not:
  - Install 14.13.0 if it's not there `nvm install 14.13.0`
  - Switch to 14.13.0 if it's not selected now `nvm use 14.13.0`
  - Try `nvm ls` again
- `npm i -g npm@6.14.8` to update to the latest npm
- `npm i` to get the latest packages
- `npm rebuild node-sass` to rebuild the site again
- `npm run build` to build all the things
- `npm run test-single` to run the tests

After `./manage.py runserver`, every part of the site should be working
- Check general pages
- Check datatables pages
- Check legal pages
- Check site admin tools

Everything should be working like normal

We'll need to check on Dev, Feature, Stage, etc to make sure the CircleCI config file works as expected

## After

After you're done, you can uninstall Node 14.13.0 with `nvm uninstall 14.13.0`, but that's probably a waste of time.

You should probably switch back to your previous version of Node, though:
- Check which versions you have `nvm ls`
- Switch back to not-14. For me, I used `nvm use 10.16.0`
- (If you don't have 10.16.0, install it with `nvm install 10.16.0`)
____

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -150,8 +150,8 @@
   },
   "browserslist": "last 2 versions, > 1%, IE 10",
   "engines": {
-    "node": "10.16.0",
-    "npm": "6.10.2"
+    "node": "14.13.0",
+    "npm": "6.14.8"
```
Comment (rfultz) [inline package.json:154]: (These are just notes for humans and aren't read by anything inside our code)

**Review by JonellaCulmer — APPROVED** [review]
Everything appears to be working and displaying as expected. Thanks @rfultz

**Comment by pkfec** [comment]
@rfultz - While testing node upgrade, i came across unapplied migration notice on my terminal. Could you please check if you feature branch needs to updated with this new migration?


(cms379) macadmins-mbp-5:fec pkasireddy$ ./manage.py runserver
Watching for file changes with StatReloader
INFO:django.utils.autoreload: Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).

```
You have 1 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): home.
Run 'python manage.py migrate' to apply them.

```
October 29, 2020 - 13:45:54
Django version 2.2.16, using settings 'fec.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.

**Comment by rfultz** [comment]
Weird. I just merged develop into it. The PR doesn't show that there were any changes to migration files, templates, or views…

> @rfultz - While testing node upgrade, i came across unapplied migration notice on my terminal. Could you please check if you feature branch needs to updated with this new migration?

**Comment by pkfec** [comment]
Looks good after you rebase your branch. Not seeing any migration related msg on my terminal. Tested CMS-API interaction, looks OK.

**Review by pkfec — APPROVED** [review]
LGTM @rfultz

**Review by johnnyporkchops — APPROVED** [review]
Works great. Tested functionality locally. Thanks for thorough, user-friendly  "How-to-test".

**Review by patphongs — APPROVED** [review]
Looks great @rfultz !

**Pull request merged**